lib/registrant.rb

```diff
@@ -0,0 +1,18 @@
+class Registrant
+  attr_reader :name, :age, :permit
+  attr_accessor :license_data
+  def initialize(name, age, permit = false, license_data = {:written=>false, :license=>false, :renewed=>false})
+    @name = name
+    @age = age
+    @permit = permit
+    @license_data = license_data
+  end
```

Comment:
I think one common thing I see with new developers is that there is some confusion surrounding when something should be set as a default parameter vs a default value in the `initialize` method. One way to remember this is to think about how many potential arguments are going to be passed to the method in question. 

In this example, there will never be an instance where license_data will be set to anything other than `{ :written => false, :license => false, :renewed => false }`, never. And since this is the case this should be set at the default within the method.

This is different than `permit` since there will be instances where a new `Registrant` will be instantiated with a value of `true` as the 3rd optional argument.

```ruby
def initialize(name, age, permit = false)
  @name = name
  @age = age
  @permit = permit
  @license_data = { :written => false, :license => false, :renewed => false }
end
```

One other thing, since you are using Symbols as keys, you do not need the hash rocket.

```ruby
@license_data = { written: false, license: false, renewed: false }
```

Lastly, you've defined an `attr_writer` for `@license_data`, which allows direct modification of the license_data hash from outside the class. This is only necessary if you want to allow `@license_data` to point to something other than the hash it is currently pointing to.

Because `@license_data` is pointing to a hash and hashes can be updated without the `attr_writer`, line 3 can be removed.